analysis/analysis_utils.py

```diff
@@ -108,14 +111,277 @@ def __init__(self):
+            use_images = patch_visit_ids
+        else:
+            visit_only = np.where(((patch_visit_times > mjd_lims[0])
+                                   & (patch_visit_times < mjd_lims[1])))[0]
+            print(visit_only)
+            use_images = patch_visit_ids[visit_only]
+
+        image_params['mjd'] = np.array([image_time_dict[str(visit_id)] for visit_id in use_images])
+        times = image_params['mjd'] - image_params['mjd'][0]
+
+        flags = ~0 # mask pixels with any flags
+        flag_exceptions = [32,39] # unless it has one of these special combinations of flags
+        master_flags = int('100111', 2) # mask any pixels which have any of
+        # these flags in more than two images
+
+        hdulist = fits.open('%s/%s' % (im_filepath, self.return_filename(use_images[0])))
+        w = WCS(hdulist[1].header)
+        image_params['ec_angle'] = self.calc_ecliptic_angle(w)
+        del(hdulist)
+
+        images = [kb.layered_image('%s/%s' % (im_filepath, self.return_filename(f))) for f in np.sort(use_images)]
+
+        print('Loaded %i images' %(len(images)))
+
+        p = kb.psf(psf_val)
+        stack = kb.image_stack(images)
+
+        # Apply masks
+        stack.apply_mask_flags(flags, flag_exceptions)
+        stack.apply_master_mask(master_flags, 2)
+
+        stack.grow_mask()
+        stack.grow_mask()
+
+        stack.apply_mask_threshold(120.)
+
+        stack.set_times(times)
+        print("Times set")
+
+        image_params['x_size'] = stack.get_width()
+        image_params['y_size'] = stack.get_width()
+
+        search = kb.stack_search(stack, p)
+
+        return(search,image_params)
+
+    def process_results(self,search,image_params,res_filepath,likelihood_level):
+        """
+        Processes results that are output by the gpu search.
+        """
+
+        keep = {'stamps': [], 'new_lh': [], 'results': [], 'times': [],
+                'lc': [], 'final_results': []}
+        likelihood_limit = False
+        res_num = 0
+        chunk_size = 500000
+        print('---------------------------------------')
+        print("Processing Results")
+        print('---------------------------------------')
+        while likelihood_limit is False:
+            pool = mp.Pool(processes=16)
+            results = search.get_results(res_num,chunk_size)
+            chunk_headers = ("Chunk Start", "Chunk Size", "Chunk Max Likelihood",
+                             "Chunk Min. Likelihood")
+            chunk_values = (res_num, len(keep['results']), results[0].lh, results[-1].lh)
+            for header, val, in zip(chunk_headers, chunk_values):
+                if type(val) == np.int:
+                    print('%s = %i' % (header, val))
+                else:
+                    print('%s = %.2f' % (header, val))
+            print('---------------------------------------')
+            psi_curves = []
+            phi_curves = []
+            print(search.get_results(0,10))
```

Comment:
Do we want to keep this or maybe set some verbose setting?

Reply:
I think we should keep this by default, but we could maybe cut down on the output in filter_results(), or at least put it behind some verbosity setting as you suggest.